app/main.py

```diff
@@ -1,25 +1,95 @@
 from abc import ABC
+from typing import Any, Type
 
 
 class IntegerRange:
-    pass
+    def __init__(
+            self,
+            min_amount: int,
+            max_amount: int
+    ) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: Any, name: Any) -> None:
+        self.name = name
```

Comment:
It should be protected name
```suggestion
        self.name = "_" + name
```